File: best time to buy and sell stock 2/sol.py
class Solution(object):
    def maxProfit(self, prices):
        buy = prices[0]
        max = 0
        size = len(prices)

        for i in range(1, size):
            if prices[i] > prices[i-1]:
                temp = prices[i] - prices[i-1]
                max = max + temp
        return max


        # Time complexity O(n)
        # Space complexity O(1)

        

File: best time to buy and sell stock 2/sol.java
class Solution {
    public int maxProfit(int[] prices) {
        int max = 0;
        int size = prices.length;
        for(int i=1;i<size;i++)
        {
            if(prices[i] > prices[i-1])
            {
                max += prices[i] - prices[i-1];
            }
        }
        return max;
    }
}
